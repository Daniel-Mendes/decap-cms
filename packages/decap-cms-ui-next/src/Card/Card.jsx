import styled from '@emotion/styled';

const Card = styled.div`
  background-color: ${({ theme }) => theme.color.elevatedSurface};
  ${({ rounded }) =>
    rounded ? `border-radius: ${rounded === 'sm' ? 2 : rounded === 'lg' ? 6 : 4}px` : ``};
  ${({ theme, direction, elevation }) =>
    elevation &&
    `box-shadow: ${theme.shadow({
      direction,
      size: elevation && typeof elevation === 'string' ? elevation : 'xs',
      theme,
    })}
    `};
`;

Card.defaultProps = { direction: 'down', rounded: 'lg' };

export default Card;
