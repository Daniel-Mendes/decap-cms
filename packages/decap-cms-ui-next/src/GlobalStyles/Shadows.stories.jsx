import React from 'react';
import styled from '@emotion/styled';

import Card from '../Card';

export default {
  title: 'Foundations/Shadows',
  parameters: {
    options: {
      showPanel: false,
    },
  },
};

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  gap: 4rem;
`;

const ElevationWrapper = styled.div`
  display: flex;
  gap: 2rem;
`;

const StyledCard = styled(Card)`
  padding: 1rem;
`;

const PropertyList = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;
`;

const Property = styled.li`
  display: flex;
  gap: 1rem;
`;

const PropertyName = styled.p`
  font-weight: bold;
  margin: 0;
`;

const PropertyValue = styled.pre`
  font-weight: normal;
  margin: 0;
`;

export function _Shadows() {
  const elevations = ['xs', 'sm', 'md', 'lg'];
  const directions = ['up', 'down', 'left', 'right'];

  return (
    <>
      <h1> Shadows</h1>
      <Wrapper>
        {elevations.map(elevation => (
          <ElevationWrapper key={elevation}>
            {directions.map(direction => (
              <StyledCard
                key={`${elevation}-${direction}`}
                elevation={elevation}
                direction={direction}
              >
                <PropertyList>
                  <Property>
                    <PropertyName>Elevation</PropertyName>
                    <PropertyValue>{elevation}</PropertyValue>
                  </Property>
                  <Property>
                    <PropertyName>Direction</PropertyName>
                    <PropertyValue>{direction}</PropertyValue>
                  </Property>
                </PropertyList>
              </StyledCard>
            ))}
          </ElevationWrapper>
        ))}
      </Wrapper>
    </>
  );
}
