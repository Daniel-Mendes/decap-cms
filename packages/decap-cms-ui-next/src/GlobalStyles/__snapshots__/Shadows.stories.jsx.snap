// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foundations/Shadows _Shadows smoke-test 1`] = `"<h1> Shadows</h1><div class="css-e0vqb1-Wrapper e4ghpa76"><div class="css-psljlt-ElevationWrapper e4ghpa75"><div elevation="xs" direction="up" class="css-15j06vv-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">xs</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">up</pre></li></ul></div><div elevation="xs" direction="down" class="css-1u0dqkj-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">xs</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">down</pre></li></ul></div><div elevation="xs" direction="left" class="css-19nxtck-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">xs</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">left</pre></li></ul></div><div elevation="xs" direction="right" class="css-bkzhy3-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">xs</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">right</pre></li></ul></div></div><div class="css-psljlt-ElevationWrapper e4ghpa75"><div elevation="sm" direction="up" class="css-17oav5w-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">sm</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">up</pre></li></ul></div><div elevation="sm" direction="down" class="css-5s7ryg-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">sm</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">down</pre></li></ul></div><div elevation="sm" direction="left" class="css-1j24b85-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">sm</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">left</pre></li></ul></div><div elevation="sm" direction="right" class="css-1khccbt-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">sm</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">right</pre></li></ul></div></div><div class="css-psljlt-ElevationWrapper e4ghpa75"><div elevation="md" direction="up" class="css-951brl-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">md</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">up</pre></li></ul></div><div elevation="md" direction="down" class="css-twb7i9-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">md</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">down</pre></li></ul></div><div elevation="md" direction="left" class="css-pn9qb7-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">md</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">left</pre></li></ul></div><div elevation="md" direction="right" class="css-c1cmjd-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">md</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">right</pre></li></ul></div></div><div class="css-psljlt-ElevationWrapper e4ghpa75"><div elevation="lg" direction="up" class="css-rs6xfh-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">lg</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">up</pre></li></ul></div><div elevation="lg" direction="down" class="css-l2kdy9-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">lg</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">down</pre></li></ul></div><div elevation="lg" direction="left" class="css-13wegqm-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">lg</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">left</pre></li></ul></div><div elevation="lg" direction="right" class="css-1jn5101-Card-StyledCard e4ghpa74"><ul class="css-1dlzwwr-PropertyList e4ghpa73"><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Elevation</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">lg</pre></li><li class="css-1mqariz-Property e4ghpa72"><p class="css-jtgqb6-PropertyName e4ghpa71">Direction</p><pre class="css-1ik4qsu-PropertyValue e4ghpa70">right</pre></li></ul></div></div></div>"`;
